WordPress/gutenberg
# Use `metadata.name` only as the hint for pattern overrides

## What?

One possible solution for https://github.com/WordPress/gutenberg/issues/58601.

Use only `metadata.name` as the "hint" for pattern overrides. Instead of requiring the `bindings` attribute for each bound attribute, we instead rely on only the existence of the `metadata.name` attribute to enable pattern overrides. This also means we hardcode and maintain a list of supported attributes for each supported block.

> [!Note]
> One thing to consider is that block renaming API will land in 6.5 but pattern overrides will not. Using names as the only requirement for enabling pattern overrides might surprise the users in some way. We have to figure out the UX about this.

## Why?

See https://github.com/WordPress/gutenberg/issues/58601.

## How?

Remove the need for `bindings` for pattern overrides.

## Testing Instructions




CI should pass.

1. Create a synced pattern and provide names for some blocks (supported blocks are paragraph, heading, button, and image.)
2. Create a post and insert the pattern.
3. Edit the blocks that have names.
4. Expect the frontend to reflect the changes too.

## Screenshots or screencast 
TBD
